### PR TITLE
UMS - Create Users CRUD

### LINKED ISSUE #39: UMS - Create Users CRUD
Implement UserDataSource and UserRepository;
Create endpoints for User CRUD

### PR DESCRIPTION
**Pull request by Igasus**
Closes #39 
(feat): Added UserController with all CRUD operations.

### CODE REVIEW

File: Server/UserManagement/UserManagement.Application/CQRS/QueryHandlers/GetUserByIdQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    private readonly IUserDataSource _dataSource;
+
+    public GetUserByIdQueryHandler(IUserDataSource dataSource)
+    {
+        _dataSource = dataSource;
+    }
+
+    public async Task<GetUserByIdQueryResponse> Handle(GetUserByIdQuery request, CancellationToken cancellationToken)
+    {
+        var user = await _dataSource.GetByIdAsync(request.UserId);
```
Comment (Vladislawv) [inline Server/UserManagement/UserManagement.Application/CQRS/QueryHandlers/GetUserByIdQueryHandler.cs]: you can pass cancellation token here

File: Server/UserManagement/UserManagement.Application/CQRS/QueryHandlers/GetUsersQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private readonly IUserDataSource _dataSource;
+
+    public GetUsersQueryHandler(IUserDataSource dataSource)
+    {
+        _dataSource = dataSource;
+    }
+
+    public async Task<GetUsersQueryResponse> Handle(GetUsersQuery request, CancellationToken cancellationToken)
+    {
+        var users = await _dataSource.GetAllAsync();
```
Comment (Vladislawv) [inline Server/UserManagement/UserManagement.Application/CQRS/QueryHandlers/GetUsersQueryHandler.cs]: and here

**Review by Vladislawv — APPROVED** [review]

File: Server/UserManagement/UserManagement.Infrastructure/DataSources/UserDataSource.cs [hunk 3]
```diff
@@ -15,13 +18,96 @@ public UserDataSource(IDriver driver)
...
+            {
+                var userAsJson = JsonSerializer.Serialize(queryResultItem.Values);
+                var user = JsonSerializer.Deserialize<User>(userAsJson);
+                users.Add(user);
+            }
+        }
+        finally
+        {
+            await session.CloseAsync();
+        }
```
Comment (Vladislawv) [inline Server/UserManagement/UserManagement.Infrastructure/DataSources/UserDataSource.cs]: I think it'll be fine to get the session by public method of the IDisposable class, that will incapsulate connection management, such a CloseAsync etc.

Comment (Igasus) [reply Server/UserManagement/UserManagement.Infrastructure/DataSources/UserDataSource.cs]: Great idea! <3

**Pull request merged**